tests/lib/rules/prefer-query-by-disappearance.test.ts

```diff
@@ -0,0 +1,749 @@
+          screen.queryByText("hello")
+        })
+      `,
+    },
+    {
+      code: `
+        import { screen } from '@testing-library/react';
+
+        await waitForElementToBeRemoved(function() {
+          return screen.queryByText('hey')
+        })
+      `,
+    },
+    {
+      code: `
+        import { screen } from '@testing-library/react';
+
+        await waitForElementToBeRemoved(function() {
+          otherCode()
+          return screen.queryByText('hey')
+        })
+      `,
+    },
+    {
+      code: `
+        import { screen } from '@testing-library/react';
+
+        await waitForElementToBeRemoved(screen.queryByText("hello"))
+      `,
+    },
+    {
+      code: `
+        import { screen } from '@testing-library/react';
+
+        const { queryByText } = screen
+        await waitForElementToBeRemoved(queryByText("hello"))
+      `,
+    },
+    {
+      code: `
+        import { screen } from '@testing-library/react';
+
+        const { queryByText } = screen
+        await waitForElementToBeRemoved(() => queryByText("hello"))
+      `,
+    },
+    {
+      code: `
+        import { screen } from '@testing-library/react';
+
+        const { queryByText } = screen
+        await waitForElementToBeRemoved(() => {
+          queryByText("hello")
+        })
+      `,
+    },
+    {
+      code: `
+        import { screen } from '@testing-library/react';
+
+        const { queryByText } = screen
+        await waitForElementToBeRemoved(() => {
+          return queryByText("hello")
+        })
+      `,
+    },
+    {
+      code: `
+        import { render } from '@testing-library/react';
+
+        const { queryByText } = render(<App />)
+        await waitForElementToBeRemoved(() => {
+          return queryByText("hello")
+        })
+      `,
+    },
+    {
+      code: `
+        import { render } from '@testing-library/react';
+
+        const { queryByText } = render(<App />)
+        await waitForElementToBeRemoved(() => {
+          queryByText("hello")
+        })
+      `,
+    },
+    {
+      code: `
+        import { render } from '@testing-library/react';
+
+        const { queryByText } = render(<App />)
+        await waitForElementToBeRemoved(() => queryByText("hello"))
+      `,
+    },
+    {
+      code: `
+        import { render } from '@testing-library/react';
+
+        const { queryByText } = render(<App />)
+        await waitForElementToBeRemoved(queryByText("hello"))
+      `,
+    },
+    {
+      code: `
+        import { render } from '@testing-library/react';
+
+        const { queryByText } = render(<App />)
+        await waitForElementToBeRemoved(function() {
+          queryByText("hello")
+        })
+      `,
+    },
+    {
+      code: `
+        import { render } from '@testing-library/react';
+
+        const { queryByText } = render(<App />)
+        await waitForElementToBeRemoved(function() {
+          return queryByText("hello")
+        })
+      `,
+    },
+  ],
```

Comment:
would a scenario like (simplified) this work with the rule?

```ts
const callback = () => screen.queryByText('text')
await waitForElementToBeRemoved(callback)
```

If it works, would you mind adding some scenarios to test it?
If it doesn't, if it's not too much trouble we could add them. If it is complex, maybe we could tackle it in another issue. I think some rules check this kind of scenarios

Reply:
This is an interesting case, but I'm happy with the current implementation so it's ok for me if we want to tackle that in a separate PR if it's too much work to implement it now.

Reply:
Seems like the same case as https://github.com/testing-library/eslint-plugin-testing-library/pull/402#discussion_r663524367

I am not sure where to start with this one, considering the callback might be imported from elsewhere 🤔 Anywho, I agree that it shouldn't be a blocker to merge this rule, as both of you mentioned.